tensorflow/core/kernels/norm_ops.cc

```diff
@@ -0,0 +1,69 @@
+
+#define EIGEN_USE_THREADS
+
+#include "tensorflow/core/kernels/norm_ops.h"
+#include "tensorflow/core/framework/op.h"
+#include "tensorflow/core/framework/op_kernel.h"
+#include "tensorflow/core/kernels/bounds_check.h"
+#include "tensorflow/core/framework/register_types.h"
+#include "tensorflow/core/kernels/fill_functor.h"
+
+namespace tensorflow {
+typedef Eigen::ThreadPoolDevice CPUDevice;
+
+namespace functor {
+
+template <typename T>
+struct Norm1D<CPUDevice, T> {
+   void operator()(const CPUDevice& d, typename TTypes<T>::ConstFlat x,
+		   typename TTypes<T>::ConstFlat mean,
+		   typename TTypes<T>::ConstFlat stdd,
+		   typename TTypes<T>::Flat out) {
+	   auto new_x = x - mean;
+	   auto z = - new_x.square() / (stdd.constant(2.0) * stdd.square());
```

Comment:
I would force the 2.0 constant to be of the same type as T by writing the code as:
   stdd.constant(static_cast<T>(2.0))